sdk/clientcore/core/src/main/java/io/clientcore/core/implementation/instrumentation/otel/OTelInstrumentation.java

```diff
@@ -176,8 +182,40 @@ public <T> InstrumentationContext createInstrumentationContext(T context) {
         return OTelSpanContext.getInvalid();
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean shouldInstrument(SpanKind spanKind, InstrumentationContext context) {
+        if (!isTracingEnabled && !isMetricsEnabled) {
+            return false;
+        }
+
+        if (allowNestedSpans) {
+            return true;
+        }
+
+        return spanKind != tryGetSpanKind(context);
```

Comment:
Same question, where should `tryGetSpanKind` returning null mean instrumentation should happen.

Reply:
yep, if there is no existing span in the parent context, the instrumentation should happen. Only if there is a span with the same kind it should not.